agent/agent.go

```diff
@@ -92,6 +99,77 @@ func (agent *Agent) RegisterServer() error {
 	return nil
 }
 
+func (agent *Agent) PerformUpgrade() error {
+	package_list, err := agent.client.FetchUpgradeablePackages()
+
+	if err != nil {
+		log.Info("Can't fetch upgrade info: %s", err)
+		return err
+	}
+
+	if agent.server.DebianLike() {
+		agent.runDebianUpgrade(package_list)
+	} else {
+		log.Fatal("Sorry, we don't support your operating system at the moment. Is this a mistake? Run `appcanary detect-os` and tell us about it at support@appcanary.com")
+	}
+	return nil
+}
+
+func (agent *Agent) runDebianUpgrade(package_list map[string]string) {
+	updateCmd := "apt-get"
+	updateArg := []string{"update", "-q"}
+
+	installCmd := "apt-get"
+	installArg := []string{"install", "--only-upgrade", "--no-install-recommends", "-y", "-q"}
+
+	for name, version := range package_list {
+		installArg = append(installArg, name+"="+version)
+	}
+
+	if env.DryRun {
+		log.Info("Running upgrade in dry-run mode...")
+	}
+
+	err := agent.runCmd(updateCmd, updateArg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = agent.runCmd(installCmd, installArg)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (agent *Agent) runCmd(cmd_name string, args []string) error {
+	_, err := exec.LookPath(cmd_name)
+
+	if err != nil {
+		log.Fatal("Can't find " + cmd_name)
+	}
+
+	cmd := exec.Command(cmd_name, args...)
+
+	var output bytes.Buffer
+	cmd.Stdout = &output
+	cmd.Stderr = &output
+
+	log.Infof("Running: %s %s", cmd_name, strings.Join(args, " "))
```

Comment:
In practice tho it won't quit look that way. Here's a sample output:

```
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ sudo CANARY_ENV=test ./appcanary upgrade -server server.conf  -conf test.conf -url http://192.168.2.12:4567 --dry-run

        ********** ********  ******** **********
       /////**/// /**/////  **////// /////**///
           /**    /**      /**           /**
           /**    /******* /*********    /**
           /**    /**////  ////////**    /**
           /**    /**             /**    /**
           /**    /******** ********     /**
           //     //////// ////////      //

2016-06-13T22:09:04.105Z 2992 conf.go:49] Found, read server conf.
2016-06-13T22:09:04.114Z 2992 client.go:167] Request: GET http://192.168.2.12:4567/api/v1/agent/servers/123456
2016-06-13T22:09:04.215Z 2992 agent.go:130] Running upgrade in dry-run mode...
2016-06-13T22:09:04.216Z 2992 agent.go:157] Running: apt-get update -q
2016-06-13T22:09:04.216Z 2992 agent.go:157] Running: apt-get install --only-upgrade --no-install-recommends -y -q libkrb5-3=1.12+dfsg-2ubuntu5.2 isc-dhcp-client=4.2.4-7ubuntu12.4
```

I think the "Running" label is necessary to preserve context.